In [20]:
import pandas as pd
import numpy as np
data=pd.read_csv('processed_nodummies.csv', index_col=0)
data.shape

(121, 54)

In [21]:

data.corr()

Student_Count_Total  \
Student_Count_Total                                      1.000000   
Average_ACT_School                                       0.511578   
College_Enrollment_Rate_School                           0.382159   
Graduation_Rate_School                                   0.394248   
Growth_ACT_Grade_11_Pct                                  0.258348   
Attainment_ACT_Grade_11_Pct                              0.516508   
Freshmen_On_Track_School_Pct_Year_2                      0.197958   
Graduation_5_Year_School_Pct_Year_2                      0.394248   
College_Enrollment_School_Pct_Year_2                     0.382159   
College_Persistence_School_Pct_Year_2                    0.514897   
lib_cnt                                                 -0.171297   
PRED_NON_ENG_PERCENT                                     0.057571   
PERCENT OF HOUSING CROWDED                              -0.109958   
PERCENT HOUSEHOLDS BELOW POVERTY                        -0.452239   
PERCENT AGED 16+ UNEMPLOYED                             -0.348923   
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA            -0.153515   
PERCENT AGED UNDER 18 OR OVER 64                        -0.133243   
PER CAPITA INCOME                                        0.296633   
HARDSHIP INDEX                                          -0.305674   
Suspensions_Per_100_Students_Change                      0.015912   
Misconducts_To_Suspensions_Change                       -0.228568   
Student_Attendance_Change                               -0.100226   
Teacher_Attendance_Change                                0.160758   
One_Year_Dropout_Rate_Change                             0.023408   
Student_Low_Income_Pct                                  -0.516919   
Student_Special_Ed_Pct                                  -0.405661   
Student_English_Learners_Pct                            -0.005903   
Student_Black_Pct                                       -0.442892   
Student_Hispanic_Pct                                     0.257746   
Student_White_Pct                                        0.519923   
Student_Asian_Pct                                        0.451429   

                                              Average_ACT_School  \
Student_Count_Total                                     0.511578   
Average_ACT_School                                      1.000000   
College_Enrollment_Rate_School                          0.681003   
Graduation_Rate_School                                  0.457470   
Growth_ACT_Grade_11_Pct                                 0.764274   
Attainment_ACT_Grade_11_Pct                             0.976073   
Freshmen_On_Track_School_Pct_Year_2                     0.441153   
Graduation_5_Year_School_Pct_Year_2                     0.457470   
College_Enrollment_School_Pct_Year_2                    0.681003   
College_Persistence_School_Pct_Year_2                   0.754701   
lib_cnt                                                -0.020359   
PRED_NON_ENG_PERCENT                                   -0.103465   
PERCENT OF HOUSING CROWDED                             -0.226397   
PERCENT HOUSEHOLDS BELOW POVERTY                       -0.377372   
PERCENT AGED 16+ UNEMPLOYED                            -0.418882   
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA           -0.324153   
PERCENT AGED UNDER 18 OR OVER 64                       -0.402597   
PER CAPITA INCOME                                       0.567241   
HARDSHIP INDEX                                         -0.470728   
Suspensions_Per_100_Students_Change                     0.010552   
Misconducts_To_Suspensions_Change                      -0.153501   
Student_Attendance_Change                              -0.079097   
Teacher_Attendance_Change                              -0.075312   
One_Year_Dropout_Rate_Change                           -0.020547   
Student_Low_Income_Pct                                 -0.785647   
Student_Special_Ed_Pct                                 -0.419518   
S

Graduation rate appears to have a strong correlation with College_Enrollment_Rate_School. Student_Special_Ed_Pct has strong negative correlation with graduation rate.

Couple variables look like duplicates: Graduation_5_Year_School_Pct_Year_2 (duplicate of Graduation_Rate_School),College_Enrollment_School_Pct_Year_2 (duplicate of College_Enrollment_Rate_School). Those variable will have to be dropped.

Few variable have weak correlation with graduation rate: College_Persistence_School_Pct_Year_2 (0.47), Attainment_ACT_Grade_11_Pct (0.45), Growth_ACT_Grade_11_Pct (0.34), Average_ACT_School (0.46), Student_Count_Total (0.39).